addons/account/models/chart_template.py

```diff
@@ -398,6 +398,15 @@ def _pre_load_data(self, template_code, company, template_data, data):
             if model in data:
                 data[model] = data.pop(model)
 
+        # Remove data of unknown fields present in the company template
+        filter_not_defined_yet = lambda key: key in company._fields or self.env.context.get('l10n_check_fields_complete')
+
+        company_data = data.get('res.company')
+        if company_data:
+            for field in list(company_data.get(company.id)):
+                if not filter_not_defined_yet(field):
+                    del data['res.company'][company.id][field]
```

Comment:
I don't think the lambda function makes it easier to read
```suggestion
        company_data = data.get('res.company')
        if company_data and not self.env.context.get('l10n_check_fields_complete'):
            for fname in list(company_data.get(company.id)):
                if fname not in company._fields:
                    del data['res.company'][company.id][field]
```